opencog/reasoning/pln/rules/not.scm

```diff
@@ -8,39 +8,33 @@
 ;    A
 ;----------------------------------------------------------------------
 
-; Test NotLink rule. For now the TV formula assumes the concept is
-; flat, or constant, in other words it's just fuzzy values, we ignore
-; the confidence for now.
-
 (define pln-rule-not
   (BindLink
     (VariableList
-      (VariableNode "$A")
-      (TypedVariableLink
-       (VariableNode "$A")
-       (TypeNode "PredicateNode")))
+      (VariableNode "$A"))
     (ImplicationLink
       (VariableNode "$A")
       (ExecutionOutputLink
         (GroundedSchemaNode "scm: pln-formula-not")
         (ListLink
+		  (NotLink
+			  (VariableNode "$A"))
           (VariableNode "$A"))))))
 
-(define (pln-formula-not A)
-  (cog-set-tv!
-   (NotLink A)
-   (pln-formula-not-side-effect-free A))
-)
+(define (pln-formula-not nA A)
+  (if (eq? (cog-type A) 'NotLink)
+	(cog-set-tv!
+		(gar A)
+		(pln-formula-not-side-effect-free A))
+	(cog-set-tv!
+		nA
+		(pln-formula-not-side-effect-free A))))
 
 (define (negate x)
```

Comment:
rename the function to negate-strength, as there is a negate function in guile https://www.gnu.org/software/guile/manual/guile.html#Procedure-Properties